lib/internal/mime.js

```diff
@@ -144,24 +251,33 @@ class MIMEParams {
     const data = this.#data;
     name = `${name}`;
     value = `${value}`;
-    const invalidNameIndex = SafeStringPrototypeSearch(name, NOT_HTTP_TOKEN_CODE_POINT);
-    if (name.length === 0 || invalidNameIndex !== -1) {
+    let invalidNameIndex = -1;
```

Comment:
maybe better to make the obvious first

```js
  set(name, value) {
    const data = this.#data;
    name = `${name}`;
    value = `${value}`;

    // String.prototype.search is needed if we want to
    // explicitly indicate to the end-user where the error is located
    if (name.length === 0) {
      throw new ERR_INVALID_MIME_SYNTAX(
        'parameter name',
        name,
        -1,
      );
    }
    
    if (value.length > 0) {
      throw new ERR_INVALID_MIME_SYNTAX(
        'parameter value',
        value,
        -1,
      );
    }

    let invalidNameIndex = -1;
    if ((invalidNameIndex = SafeStringPrototypeSearch(name, NOT_HTTP_TOKEN_CODE_POINT)
      , invalidNameIndex) !== -1
    ) {
      throw new ERR_INVALID_MIME_SYNTAX(
        'parameter name',
        name,
        invalidNameIndex,
      );
    }
    let invalidValueIndex = -1;
    if ((invalidValueIndex = SafeStringPrototypeSearch(value, NOT_HTTP_QUOTED_STRING_CODE_POINT),
      invalidValueIndex) !== -1
    ) {
      throw new ERR_INVALID_MIME_SYNTAX(
        'parameter value',
        value,
        invalidValueIndex,
      );
    }

    data.set(name, value);
  }
```

Reply:
Actually. Do we need to do the comma?
The assignment (invalidNameIndex = SafeStringPrototypeSearch(name, NOT_HTTP_TOKEN_CODE_POINT) returns the value. 

so we could reduce it to this:
```js
if ((invalidNameIndex = SafeStringPrototypeSearch(name, NOT_HTTP_TOKEN_CODE_POINT)) !== -1) {
```